src/hidimstat/noise_std.py

```diff
@@ -5,283 +5,256 @@
     -------
-    cov_hat : ndarray, shape (n_times, n_times)
-        Estimated covariance matrix.
+    sigma_hat/cov_hat : float/ndarray, shape (n_times, n_times)
+        Estimated noise standard deviation based on residuals
+        or estimated covariance matrix for group.
 
-    beta_hat : ndarray, shape (n_features, n_times)
-        Estimated parameter matrix.
+    beta_hat : ndarray, shape (n_features,)/(n_features, n_times)
+        Estimated sparse coefficient vector from Lasso regression.
 
     References
     ----------
-    .. [1] Chevalier, J. A., Gramfort, A., Salmon, J., & Thirion, B. (2020).
-           Statistical control for spatio-temporal MEG/EEG source imaging with
-           desparsified multi-task Lasso. In NeurIPS 2020-34h Conference on
-           Neural Information Processing Systems.
+    .. footbibliography::
     """
 
-    X = np.asarray(X)
-    n_samples, n_features = X.shape
-    n_times = Y.shape[1]
-
-    if method == "simple":
-        print("Group reid: simple cov estimation")
-    else:
-        print(f"Group reid: {method}{order} cov estimation")
-
-    if (max_iter // 5) <= n_features:
-        max_iter = n_features * 5
-        print(f"'max_iter' has been increased to {max_iter}")
-
-    cv = KFold(n_splits=5, shuffle=True, random_state=seed)
-
+    X_ = np.asarray(X)
+    n_samples, n_features = X_.shape
+    if group:
+        n_times = y.shape[1]
     if fit_Y:
-
-        clf_mtlcv = MultiTaskLassoCV(
+        # check if max_iter is large enough
+        if max_iter // n_split <= n_features:
+            max_iter = n_features * n_split
+            print(f"'max_iter' has been increased to {max_iter}")
+
+        # use the cross-validation for define the best alpha of Lasso
+        cv = KFold(n_splits=n_split, shuffle=True, random_state=seed)
+        Refit_CV = MultiTaskLassoCV if group else LassoCV
+        clf_cv = Refit_CV(
             eps=eps,
             fit_intercept=False,
             cv=cv,
             tol=tol,
             max_iter=max_iter,
             n_jobs=n_jobs,
         )
+        clf_cv.fit(X_, y)
 
-        clf_mtlcv.fit(X, Y)
-        beta_hat = clf_mtlcv.coef_
-        residual = clf_mtlcv.predict(X) - Y
-        row_max = np.max(np.sum(np.abs(beta_hat), axis=0))
-        support = np.sum(np.sum(np.abs(beta_hat), axis=0) > tol * row_max)
+        # Estimate the support of the variable importance
+        beta_hat = clf_cv.coef_
+        residual = clf_cv.predict(X_) - y
 
-        # avoid dividing by 0
-        support = min(support, n_samples - 1)
+        # get the number of non-zero coefficients
+        coef_ = (
+            np.sum(np.abs(beta_hat), axis=0)
+            if len(beta_hat.shape) > 1
+            else np.abs(beta_hat)
+        )
+        size_support = np.sum(coef_ > tol * coef_.max())
 
+        # avoid dividing by 0
+        size_support = min(size_support, n_samples - 1)
     else:
-
+        # null model
+        # TODO Why do we need a null model?
```

Comment:
if it's used nowhere, let's get rid of it.

Reply:
See PR #127